src/set.cr

```diff
@@ -152,6 +152,30 @@ struct Set(T)
     set
   end
 
+  # Difference: returns a new set containing elements in this set that are not
+  # present in the other.
+  #
+  #     Set.new([1,2,3,4,5]) - Set.new([2,4])               #=> Set{1, 3, 5}
+  #     Set.new(['a','b','b','z']) - Set.new(['a','b','c']) #=> Set{'z'}
+  def -(other : Set)
+    set = Set(T).new
+    each do |value|
+      set.add value unless other.includes?(value)
+    end
+    set
+  end
+
+  # Symmetric Difference: returns a new set `(self - other) | (other - self)`.
+  # Equivalently, returns `(self | other) - (self & other)`.
+  #
+  #     Set.new([1,2,3,4,5]) ^ Set.new([2,4,6])             #=> Set{1, 3, 5, 6}
+  #     Set.new(['a','b','b','z']) ^ Set.new(['a','b','c']) #=> Set{'z', 'c'}
+  def ^(other : Set(U))
+    set = self | other
+    set = set - (self & other)
+    set
+  end
```

Comment:
This creates 3 new sets instead of just one, wouldn't it be faster to do something like

```
set = Set(T).new
each do |item|
  set.add item unless other.includes? item
end
other.each do |item|
  set.add item unless includes? item
end
```
?